Station/Station.js

```diff
@@ -22,40 +29,58 @@ class Station {
     this.hasSteplessAccess = station.hasSteplessAccess;
     this.hasMobilityService = station.hasMobilityService;
     this.federalState = station.federalState;
+
     const area = station.regionalbereich;
-    this.regionalArea = new RegionalArea(area.number, area.name, area.shortName);
+    if (area) {
+        this.regionalArea = new RegionalArea(area.number, area.name, area.shortName);
+    }
+
     const adress = station.mailingAddress;
     this.mailingAddress = new MailAddress(adress.city, adress.zipcode, adress.street);
+
     const contact = station.aufgabentraeger;
+    if (contact) {
     this.aufgabentraeger = new StationContact(contact.name, contact.shortName,
       contact.email, contact.number, contact.phoneNumber);
+    }
+
     const timeTableContact = station.timeTableOffice;
-    if (timeTableContact){
+    if (timeTableContact) {
       this.timeTableOffice = new StationContact(timeTableContact.name, timeTableContact.shortName,
         timeTableContact.email, timeTableContact.number, timeTableContact.phoneNumber);
     }
+
     const szentraleContact = station.szentrale;
-    this.szentrale = new StationContact(szentraleContact.name, szentraleContact.shortName,
+    if (szentraleContact) {
+        this.szentrale = new StationContact(szentraleContact.name, szentraleContact.shortName,
       szentraleContact.email, szentraleContact.number, szentraleContact.publicPhoneNumber);
+    }
+
     const stationManagementContact = station.stationManagement;
-    this.stationManagement = new StationContact(stationManagementContact.name, stationManagementContact.shortName,
+    if (stationManagementContact) {
+        this.stationManagement = new StationContact(stationManagementContact.name, stationManagementContact.shortName,
       stationManagementContact.email, stationManagementContact.number, stationManagementContact.phoneNumber);
+    }
+
     if (station.DBinformation) {
       this.DBInformationOpeningTimes = new OpeningTimes(station.DBinformation.availability);
     }
+
     if (station.localServiceStaff) {
       this.localServiceStaffAvailability = new OpeningTimes(station.localServiceStaff.availability);
     }
+
     this.primaryEvaId = station.evaNumbers.filter(eva => eva.isMain)[0].number;
```

Comment:
It is possible, that a station does not have any evaNumbers. Therefore you have to check for existence of `[0]`. Eg. `this.primaryEvaId = access(station.evaNumbers.filter(eva => eva.isMain)[0],'number');`